### PR TITLE
replaced invalid attributes with new data API (closes #21)

### LINKED ISSUE #21: fix all invalid attribute calls in .js files
customize.js sets a non-conforming "open" attribute. It should use `data-open` instead + HTML data API.

### PR DESCRIPTION
**Pull request by sarracini**
custom attributes now use html5 data api

### CODE REVIEW

File: css/newTab.css [hunk 1]
```diff
@@ -251,11 +251,11 @@ input[type=button] {
   left: auto;
 }
 
-.newtab-site:-moz-any([type=enhanced], [type=sponsored], [suggested]) .newtab-sponsored {
+.newtab-site:-moz-any([data-type=enhanced], [data-type=sponsored], [data-suggested=true]) .newtab-sponsored {
```
Comment (marcoscaceres) [inline css/newTab.css:258]: Do type "organic", or other types need to match here too?

File: css/newTab.inc.css [hunk 2]
```diff
@@ -155,14 +155,14 @@
...
 }
 
-.newtab-site[type=affiliate] .newtab-thumbnail,
-.newtab-site[type=enhanced] .newtab-thumbnail,
-.newtab-site[type=organic] .newtab-thumbnail,
-.newtab-site[type=sponsored] .newtab-thumbnail {
+.newtab-site[data-type=affiliate] .newtab-thumbnail,
+.newtab-site[data-type=enhanced] .newtab-thumbnail,
+.newtab-site[data-type=organic] .newtab-thumbnail,
+.newtab-site[data-type=sponsored] .newtab-thumbnail {
```
Comment (marcoscaceres) [inline css/newTab.inc.css]: I think we can just collapse this to:

``` CSS
.newtab-site[data-type] .newtab-thumbnail{}
```

No?

File: js/page.js [hunk 3]
```diff
@@ -169,9 +169,9 @@
       let inputSelector = ".newtab-control, .newtab-link";
       for (let input of document.querySelectorAll(inputSelector)) {
         if (aValue) {
-          input.removeAttribute("tabindex");
+          delete input.data.tabindex;
         } else {
-          input.setAttribute("tabindex", "-1");
+          input.data.tabindex = -1;
```
Comment (marcoscaceres) [inline js/page.js]: This is a real attribute (controls the browser's tabbing behavior), so shouldn't be "data". So this one we should leave as is.

File: js/sites.js [hunk 4]
```diff
@@ -303,10 +302,10 @@
 
     _ignoreHoverEvents(element) {
       element.addEventListener("mouseover", () => {
-        this.cell.node.setAttribute("ignorehover", "true");
+        this.cell.node.data.ignorehover = true;
```
Comment (marcoscaceres) [inline js/sites.js]: Nit: It should probably be called "ignoreHover" in JS and reflected as "data-ignore-hover" in CSS. That also matches the surrounding method.

File: js/sites.js [hunk 5]
```diff
@@ -329,11 +328,11 @@
 
     _toggleLegalText(buttonClass, explanationTextClass) {
       let button = this._querySelector(buttonClass);
-      if (button.hasAttribute("active")) {
+      if (button.getAttribute("data-active") === "false") {
```
Comment (marcoscaceres) [inline js/sites.js]: Maybe: `(!button.data.active)` ... don't get the attribute :wink:

File: css/newTab.css [hunk 6]
```diff
@@ -497,7 +497,7 @@ input[type=button] {
   right: auto;
 }
 
-#newtab-customize-panel[open="true"] {
+#newtab-customize-panel[data-open="true"] {
```
Comment (marcoscaceres) [inline css/newTab.css]: Nit: note that other "true" are not quoted. Not sure if it makes a difference, but good to be consistent.

File: css/newTab.inc.css (hunk 2 above)
Comment (sarracini) [inline css/newTab.inc.css]: Yes!

**Comment by marcoscaceres** [comment]
Just one small question left.

**Comment by marcoscaceres** [comment]
Rebased

**Comment by marcoscaceres** [comment]
R+

**Pull request merged**